src/components/Form/Widgets/FileWidget.tsx

```diff
@@ -0,0 +1,216 @@
+	const [errors, setErrors] = React.useState<string[]>([]);
+
+	const accept =
+		props.uiSchema?.['ui:options']?.accept ??
+		props.uiSchema?.value?.['ui:options']?.accept;
+	const maxSize =
+		props.uiSchema?.['ui:options']?.maxSize ??
+		props.uiSchema?.value?.['ui:options']?.maxSize;
+
+	const dataUrl = (value: OnFileReadSuccessParams | string) => {
+		return typeof value === 'string' ? value : value.dataUrl;
+	};
+
+	const isImage = (value: OnFileReadSuccessParams | string) => {
+		return dataUrl(value).startsWith('data:image/');
+	};
+
+	const onDrop = React.useCallback(
+		((acceptedFiles, fileRejections) => {
+			setErrors([]);
+
+			if (fileRejections != null && fileRejections.length > 0) {
+				const errs = fileRejections[0].errors.map(
+					({ message }) => `${fileRejections[0].file.name}: ${message}`,
+				);
+				setErrors(errs);
+				onChange();
+				return onError?.(errs);
+			}
+
+			acceptedFiles.forEach((file) => {
+				const reader = new FileReader();
+				reader.onabort = () => console.log('file reading was aborted');
+				reader.onerror = () => {
+					const err = [`Failed to upload ${file.name}`];
+					onChange();
+					setErrors(err);
+					onError?.(err);
+				};
+				reader.onloadstart = () => setIsLoading(true);
+				reader.onprogress = (event) => {
+					setLoadingPercentage(Math.floor((100 * event.loaded) / event.total));
+				};
+				reader.onload = () => {
+					if (file != null && reader.result != null) {
+						setIsLoading(false);
+						onChange({
+							dataUrl: reader.result as string,
+							uploadedFile: file,
+						});
+					}
+				};
+
+				reader.readAsDataURL(file);
+			});
+		}) as NonNullable<DropzoneOptions['onDrop']>,
+		[],
+	);
+
+	const { getRootProps, getInputProps, isDragActive } = useDropzone({
+		onDrop,
+		maxSize,
+		accept,
+		multiple: false,
+		noDrag: false,
+		disabled: isLoading,
+	});
+
+	return (
+		<>
+			{props.schema.title && <InputLabel>{props.schema.title}</InputLabel>}
+			<Box
+				sx={{
+					border: (theme) =>
+						`dashed 2px ${
+							errors.length !== 0
+								? '#F19B9E'
+								: isDragActive
+								? theme.palette.primary.main
+								: 'rgba(0, 0, 0, 0.23)'
+						}`,
+					borderRadius: '2px',
+					display: 'flex',
+					flexDirection: 'column',
+					alignItems: 'center',
+					justifyContent: 'center',
+				}}
+				style={{ minHeight: '150px' }}
+				flexDirection="column"
+				alignItems="center"
+				justifyContent="center"
+				{...getRootProps()}
+				dir={undefined}
+			>
+				<input {...getInputProps()} />
+
+				{isLoading && (
+					<Box
+						sx={{
+							position: 'absolute',
+							top: 0,
+							left: 0,
+							width: '100%',
+							height: '100%',
+							display: 'flex',
+							alignItems: 'center',
+							justifyContent: 'center',
+							background: 'rgba(255, 255, 255, 0.7)',
+						}}
+					>
+						<CircularProgress variant="determinate" value={loadingPercentage} />
+					</Box>
+				)}
```

Comment:
I think we need to do this instead of putting the CircularProgress inside the Dropzone area and giving it a bunch of extra styling: https://mui.com/material-ui/api/circular-progress/#aria

Reply:
@myarmolinsky I don't think this works out of the box sadly. AFAIK this is just for accessibilit (e.g. screen readers) stuff https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-busy

I did add it anyway, but the bunch of styling for the overlay are still necessary